### PR TITLE
Add examples for subplots_axes_and_figures

### PR DESCRIPTION
**Pull request by divyam3897**
Add descriptions for subplots_axes_and_figures
Extension to PRs for https://github.com/matplotlib/matplotlib/issues/8921

### CODE REVIEW

File: examples/subplots_axes_and_figures/subplots_adjust.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 Subplots Adjust
 ===============
 
+Example showing tuning of the subplots.
```
Comment (choldgraf) [inline examples/subplots_axes_and_figures/subplots_adjust.py]: How about "Adjusting the spacing and margins of subplots."

File: examples/subplots_axes_and_figures/subplot_toolbar.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 Subplot Toolbar
 ===============
 
+Example showing various subplots which are controlled using the toolbar.
```
Comment (jklymak) [inline examples/subplots_axes_and_figures/subplot_toolbar.py]: Suggest "Matplotlib has a toolbar available for adjusting subplot spacing"

**Comment by QuLogic** [comment]
I think both of the comments are saying that you should drop the "Example of" part, because that's kind of redundant.

File: examples/subplots_axes_and_figures/subplots_adjust.py [hunk 3]
```diff
@@ -3,6 +3,7 @@
 Subplots Adjust
 ===============
 
+Example showing the adjusting of spacing and margins of subplots.
```
Comment (jklymak) [inline examples/subplots_axes_and_figures/subplots_adjust.py]: Adjusting the spacing of margins and subplots using :func:`~matplotlib.pyplot.subplots_adjust`:

File: examples/subplots_axes_and_figures/subplot_toolbar.py [hunk 4]
```diff
@@ -3,6 +3,7 @@
 Subplot Toolbar
 ===============
 
+Example showing toolbar of matplotlib available for adjusting subplot spacing.
```
Comment (jklymak) [inline examples/subplots_axes_and_figures/subplot_toolbar.py]: Matplotlib has a toolbar available for adjusting suplot spacing.

**Comment by jklymak** [comment]
@divyam3897   Thanks so much for pointing out that these descriptions were missing!  I do think that your phrasing needs tweaking.

**Comment by divyam3897** [comment]
@jklymak done!, let me know if it's better or requires more changes :)

**Comment by jklymak** [comment]
Great!  Thanks for the PR.  Very helpful!

**Comment by tacaswell** [comment]
@meeseeksdev backport to v2.1.0-doc

**Pull request merged**